Book-Your-Place/bookyourplace
# Add public booking routes views

#### What's this PR do?
Add publicly accessible booking routes and views.

##### Background context
This is part of the work to allow Guides to share a public link to a member of the public who can then use that to create a new booking for that trip.

Subsequent work will use a Stripe form instead of the basic booking#create form.

#### Where should the reviewer start?
The clearest explanation of the changes is in: app/controllers/public/trips/bookings_controller.rb
and: config/routes.rb


#### How should this be manually tested?
Go into the console:

`rails c`

Then create a new trip:
`Trip.create(name: 'FooBar')`

note it's id / uuid.

In the browser go to:
/public/trips/#{:trip_id}/bookings/new

You should see the basic views and be able to click on all the buttons to create a booking, edit a booking, etc.

#### Screenshots
New booking:
![image](https://user-images.githubusercontent.com/1453680/46960578-c2ff8e80-d096-11e8-9dc0-39a803afeb39.png)

Booking show:

![image](https://user-images.githubusercontent.com/1453680/46960611-d6aaf500-d096-11e8-8fb5-44b0b784dff4.png)

Booking edit:
![image](https://user-images.githubusercontent.com/1453680/46960636-e62a3e00-d096-11e8-9700-df3fb3cb62a6.png)

---

#### Migrations
none

#### Additional deployment instructions
none

#### Additional ENV Vars
none

#### Follow up work
The new booking page will need to be designed with our demo as guidance, ie:

![image](https://user-images.githubusercontent.com/1453680/46960833-4c16c580-d097-11e8-9e4d-fe8669c86ff7.png)


And the form will need to post to Stripe API, then we handle the callback and redirect the user, to the next step in the process.
